File: steps/python/jobs_feeding_daily.step.py
# steps/python/jobs_feeding_daily.step.py
config = { "type":"cron", "name":"PyDailyFeeding", "cron":"30 2 * * *", "emits":[], "flows":["pets_management"] }

async def handler(ctx, logger=None):
    try:
        import sys
        import os
        sys.path.append(os.path.join(os.path.dirname(__file__), '..'))
        from services import pet_store
    except ImportError:
        print("Import error in feeding job")
        return {"success": False}
    lg = (logger or (ctx or {}).get("logger"))
    for p in pet_store.list_all():
        if p.get('status') != 'adopted':
            if lg: lg.info("Feeding reminder", {"id": p["id"], "name": p["name"]})

File: steps/typescript/jobs.feeding.daily.step.ts
// steps/typescript/jobs.feeding.daily.step.ts
import { CronConfig, Handlers } from 'motia';
import { TSStore } from './ts-store';

export const config: CronConfig = {
  type: 'cron',
  name: 'TsDailyFeeding',
  cron: '30 2 * * *',   // ≈ 07:30 IST if server is UTC
  emits: [],
  flows: ['pets_management'],
};

export const handler: Handlers['TsDailyFeeding'] = async ({ logger }) => {
  for (const p of TSStore.list()) {
    if (p.status !== 'adopted') {
      logger.info('Feeding reminder', { id: p.id, name: p.name });
    }
  }
};
